debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/KafkaConnectRecordParser.java

```diff
@@ -98,13 +118,32 @@ public Record parseRecord(Object keyObj, Object valueObj) {
             parseValueFields(value, r);
 
             return r;
-        }
-        catch (Exception ex) {
+        } catch (
+
+        Exception ex) {
             LOGGER.error("Failed to parse msg: {}", ex);
             throw new RuntimeException(ex);
         }
     }
 
+    protected void parseEventId(Struct value, Record r) {
+        r.eventId = null;
+        if (sourceType.equals("oracle") || sourceType.equals("postgresql")) {
+            // Extract transaction struct from value if it is available
+            Struct transaction = value.getStruct("transaction");
+            // Transaction metadata =>
+            // Struct{id=0200030002cf0000,total_order=1,data_collection_order=1}
+            // Convert id=0200030002cf0000,total_order=1,data_collection_order=1 to string
+            if (transaction == null) {
```

Comment:
Should we error out if there is no transaction metadata in an Event? And same with `id`, `total_order`, and `data_collection_order`.